filter/inc_dedup.go

```diff
@@ -0,0 +1,90 @@
+package filter
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"github.com/AdRoll/baker"
+)
+
+const dedupHelp = `
+Removes record with identical value using a set of provided fields as key.
```

Comment:
I'd make the word duplicate appear in the first sentence, to justify the filter name immediately. Also I'd avoid explaining too much about implementation details, but focus on usage and limitation, something along the lines of:
```suggestion
This filter removes duplicate records. A record is considered a duplicate, and is thus removed by this filter, 
if another record with the same values has already been _seen_. The comparison is performed on a user-provided list of fields (Fields setting).
```